src/stratis_cli/_actions/_top.py

```diff
@@ -760,3 +783,35 @@ def list_keys(_):
         print_table(
             ["Key Description"], sorted(key_list, key=lambda entry: entry[0]), ["<"]
         )
+
+    @staticmethod
+    def unlock_pools(_):
+        """
+        Unlock all of the encrypted pools that have been detected by the daemon
+        but are still locked.
+        """
+        from ._data import Manager
+
+        proxy = get_object(TOP_OBJECT)
+
+        pool_uuid_list = _fetch_locked_pools_property(proxy)
+        if pool_uuid_list == []:
+            raise StratisCliNoChangeError("unlock", "pools")
+
+        # This block is not covered as it would require a relatively complicated
+        # simulation in the stratisd sim_engine.
+        for uuid in pool_uuid_list:  # pragma: no cover
+            ((is_some, _), return_code, message) = Manager.Methods.UnlockPool(
```

Comment:
I'm not sure whether the boolean in the returned value is redundant or not; I'm looking at the daemon side as well, and I'll try to return to that question shortly.

Reply:
This follows the pattern that we use elsewhere for things like `add-data`. If you decide this is unnecesssary, we should go back and change a few other things too.

Reply:
We introduced that boolean during the idempotency work for 2.0.0 release and I feel sure that we would never have introduced a new field in the D-Bus API that we believed was entirely redundant. So, we felt that it would be possible to return either ```(false, vec![])``` indicating that nothing was done, or ```(true, vec![])``` indicating that something was done but it resulted in no action actually being taken. I'll have to take a deeper look to see what the situation is now. Adding blockdevs is a really tricky area, if we have another example of the same return structure, it might make it clearer what distinction is being made.

Reply:
I remember that decision and I was thinking that maybe I could change the code to follow that rule? I see what you mean that it could still be redundant because it should always be a "valid" response if the response code is 0. Does that seems like what you're seeing too?